### PR TITLE
♻️ Tidy models

### PR DESCRIPTION
**Pull request by astroash**
**Refactor Changes:**
- move Roles & Permissions methods to own model file
- move tokens methods to own model file

Notes: I've not moved `EmailService` as its interface that needs initialising. I suggest that this is refactored into a similar format as model methods that use knex, and email configuration is passed to `EmailService`

### CODE REVIEW

File: src/auth/strategies/standard/validate.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import * as Hapi from 'hapi';
 import * as Boom from 'boom';
 import { Users, Organisations } from '../../../models';
-import Roles from '../../roles';
-import Permissions from '../../permissions';
+import Roles from '../../../models/role';
+import Permissions from '../../../models/__tests__/permission';
```
Comment (eliascodes) [inline src/auth/strategies/standard/validate.ts]: Errrrr...?

File: src/models/__tests__/community_business.test.integration.ts [hunk 2]
```diff
@@ -2,13 +2,13 @@ import * as Knex from 'knex';
 import { compare } from 'bcrypt';
 import { getConfig } from '../../../config';
 import factory from '../../../tests/utils/factory';
+import moment = require('moment');
```
Comment (eliascodes) [inline src/models/__tests__/community_business.test.integration.ts]:
```suggestion
import * as moment from 'moment';
```

File: src/models/__tests__/permission.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { uniqWith, equals } from 'ramda';
```
Comment (eliascodes) [inline src/models/__tests__/permission.ts]: Why is this file in a `__tests__` directory?

File: src/models/__tests__/token.test.integration.ts [hunk 4]
```diff
@@ -0,0 +1,259 @@
+import * as Knex from 'knex';
+import * as moment from 'moment';
+import { getConfig } from '../../../config';
+import { getTrx } from '../../../tests/utils/database';
+import { Users } from '..';
+import { Tokens } from '../token';
+
+
+describe('User Model', () => {
```
Comment (eliascodes) [inline src/models/__tests__/token.test.integration.ts]: Change

File: src/models/__tests__/token.test.integration.ts [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+  beforeEach(async () => {
+    trx = await getTrx(knex);
+  });
+
+  afterEach(async () => {
+    await trx.rollback();
+  });
+
+  describe('createConfirmAddRoleToken', () => {
```
Comment (eliascodes) [inline src/models/__tests__/token.test.integration.ts]: Resolve this comment if all these tests are cut and pasted from the old user model tests without any changes (except for re-naming).

File: src/models/__tests__/volunteer.test.integration.ts [hunk 6]
```diff
@@ -1,14 +1,14 @@
 import * as Knex from 'knex';
 import { compare } from 'bcrypt';
 import { omit } from 'ramda';
+import moment = require('moment');
```
Comment (eliascodes) [inline src/models/__tests__/volunteer.test.integration.ts]:
```suggestion
import * as moment from 'moment';
```

File: src/models/token.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
+import moment = require('moment');
```
Comment (eliascodes) [inline src/models/token.ts]:
```suggestion
import * as moment from 'moment';
```

**Review by eliasmalik — CHANGES_REQUESTED** [review]

**Comment by astroash** [comment]
closing as this has been recreated in mono https://github.com/TwinePlatform/twine-main/pull/11

**Pull request closed without merging**